server/src/main/java/org/elasticsearch/gateway/PersistedClusterStateService.java

```diff
@@ -130,7 +131,7 @@
     public static final String LAST_PAGE_FIELD_NAME = "last_page";
     public static final int IS_LAST_PAGE = 1;
     public static final int IS_NOT_LAST_PAGE = 0;
-    private static final int COMMIT_DATA_SIZE = 4;
+    private static final int COMMIT_DATA_SIZE = 5;
```

Comment:
We assert when loading the data that the commit's `userData` has this many entries, which will fail in upgrades from a 4-entry to a 5-entry commit I think.